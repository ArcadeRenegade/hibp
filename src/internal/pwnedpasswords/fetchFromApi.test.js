import {
  ERR,
  RANGE_VALID,
  RANGE_INVALID,
  UNEXPECTED,
} from '../../../test/fixtures';
import pwnedPasswordRange from '../../pwnedPasswordRange';
import axiosInstance from './axiosInstance';

describe('internal (pwnedpassword): fetchFromApi', () => {
  describe('request failure', () => {
    let failboat;

    beforeAll(() => {
      failboat = axiosInstance.interceptors.request.use(() => {
        throw ERR;
      });
    });

    afterAll(() => {
      axiosInstance.interceptors.request.eject(failboat);
    });

    it('should re-throw request setup errors', () =>
      expect(pwnedPasswordRange(RANGE_VALID)).rejects.toEqual(ERR));
  });

  describe('invalid range', () => {
    it('should throw an Error with "Bad Request" response data', () =>
      expect(pwnedPasswordRange(RANGE_INVALID)).rejects.toMatchSnapshot());
  });

  describe('unexpected HTTP error', () => {
    it('should throw an Error with the response status text', () =>
      expect(pwnedPasswordRange(UNEXPECTED)).rejects.toMatchSnapshot());
  });
});
