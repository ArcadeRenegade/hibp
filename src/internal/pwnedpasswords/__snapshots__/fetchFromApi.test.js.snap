// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`internal (pwnedpassword): fetchFromApi invalid range should throw an Error with "Bad Request" response data 1`] = `[Error: The hash prefix was not in a valid format]`;

exports[`internal (pwnedpassword): fetchFromApi unexpected HTTP error should throw an Error with the response status text 1`] = `[Error: Unknown - something unexpected happened.]`;
